Source/WTF/wtf/glib/GSpanExtras.h

```diff
@@ -67,22 +67,18 @@ GMallocSpan<T, Malloc> adoptGMallocSpan(std::span<T> span)
     return adoptMallocSpan<T, Malloc>(span);
 }
 
-template<typename Malloc = GMalloc>
-GMallocSpan<char, Malloc> adoptGMallocString(char* str, size_t length)
-{
-    return adoptGMallocSpan<char, Malloc>(unsafeMakeSpan(str, length));
-}
-
-template<typename Malloc = GMalloc>
-GMallocSpan<char, Malloc> adoptGMallocString(char* str)
+template<typename T, typename Malloc = GMalloc>
+GMallocSpan<T, Malloc> dupGMallocSpan(std::span<const T> span)
 {
-    WTF_ALLOW_UNSAFE_BUFFER_USAGE_BEGIN
-    return adoptGMallocSpan<char, Malloc>(unsafeMakeSpan(str, str ? strlen(str) : 0));
-    WTF_ALLOW_UNSAFE_BUFFER_USAGE_END
+    T* newMemory = static_cast<T*>(Malloc::malloc(sizeof(T) * span.size()));
+    auto newSpan = unsafeMakeSpan(newMemory, span.size());
+    memcpySpan(newSpan, span);
+    return adoptMallocSpan<T, Malloc>(newSpan);
 }
 
 WTF_EXPORT_PRIVATE Expected<GMallocSpan<char>, GUniquePtr<GError>> gFileGetContents(CStringView);
 WTF_EXPORT_PRIVATE Expected<GMallocSpan<char*, GMallocStrv>, GUniquePtr<GError>> gKeyFileGetKeys(GKeyFile*, CStringView groupName);
+
```

Comment:
Nit: This empty line does not belong in the patch, please remove it before landing.